frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1242,6 +1409,13 @@ impl<T: Config> Pallet<T> {
 	}
 
 	fn do_elect() -> Result<(Supports<T::AccountId>, Weight), ElectionError> {
+		// We have to unconditionally try finalizing the signed phase here. There are only two
+		// possibilities:
+		//
+		// - signed phase was open, in which case this is essential for correct functioning of the system
```

Comment:
Is it correct to understand that `finalize_signed_phase` is always intended to be called correctly in `on_initialize` and this is just an extra safeguard? Or is there a logic branch we know where `finalize_signed_phase` is not called in `on_initialize`?

Reply:
We predict the time remaining to the election in `on_initialize`, and try to progress the state there. However, we can't really prevent someone from calling `elect` on an entirely arbitrary block. Calling it here ensures that signed things are handled appropriately in that case.